### PR TITLE
Bug fix for x1143

### PR DESCRIPTION
**Pull request by seenanair**
- Display of  Flagged labware section  for events that is not associated with SGPs
- Removed Flagged labware section from labware details page

### CODE REVIEW

**Review by sabrine33 — APPROVED** [review]

File: src/components/history/History.tsx [hunk 1]
```diff
@@ -20,13 +20,19 @@ import { HistoryUrlParams } from '../../pages/History';
 /**
  * Component for looking up and displaying the history of labware and samples
  */
-export default function History(props: HistoryUrlParams) {
+type HistoryProps = HistoryUrlParams & { displayFlaggedLabware?: boolean };
+export default function History(props: HistoryProps) {
   const historyMachine = React.useMemo(() => {
     return createHistoryMachine();
   }, []);
+  const getHistoryURLPramas = (props: HistoryProps): HistoryUrlParams => {
```
Comment (khelwood) [inline src/components/history/History.tsx:28]: `Pramas` -> `Params`

**Review by khelwood — COMMENTED** [review]
Which bit of this solves the problem where the flagged-box was not showing up when it should?

**Comment by seenanair** [comment]
> Which bit of this solves the problem where the flagged-box was not showing up when it should?

The issue was that the code, which displays the flagged labware section (lines 263 to 289), was within the condition uniqueWorkNumbers > 0. This condition would fail for events like 'Flag labware,' which is not associated with any SGP's. Therefore, it has been taken out of that condition. I hope this makes sense.

File: src/components/history/History.tsx (hunk 1 above)
Comment (seenanair) [reply src/components/history/History.tsx]: Fixed

File: src/components/history/History.tsx [hunk 2]
```diff
@@ -20,13 +20,19 @@ import { HistoryUrlParams } from '../../pages/History';
 /**
  * Component for looking up and displaying the history of labware and samples
  */
-export default function History(props: HistoryUrlParams) {
+type HistoryProps = HistoryUrlParams & { displayFlaggedLabware?: boolean };
+export default function History(props: HistoryProps) {
   const historyMachine = React.useMemo(() => {
     return createHistoryMachine();
   }, []);
+  const getHistoryURLParmas = (props: HistoryProps): HistoryUrlParams => {
```
Comment (khelwood) [inline src/components/history/History.tsx:28]: `Parmas` -> `Params`

**Review by khelwood — APPROVED** [review]

**Pull request merged**